alternator/conditions.cc

```diff
@@ -232,7 +232,14 @@ bool check_BEGINS_WITH(const rjson::value* v1, const rjson::value& v2,
     if (it2->name == "S") {
         return rjson::to_string_view(it1->value).starts_with(rjson::to_string_view(it2->value));
     } else /* it2->name == "B" */ {
-        return base64_begins_with(rjson::to_string_view(it1->value), rjson::to_string_view(it2->value));
+        try {
+            return base64_begins_with(rjson::to_string_view(it1->value), rjson::to_string_view(it2->value));
+        } catch(...) {
```

Comment:
Can you catch the specific exception you expect instead of every type?

Reply:
yes. Is it because we can also return some kind of out of memory exception or similar indicating that's internal error instead of user error?

Reply:
Yes. Maybe in the future (or already today, I didn't verify) this function may throw some other error (like out of memory) which doesn't mean it's an unparsable string, and shouldn't be swallowed as we do today?

If you're sure that base64_begins_with is guaranteed to only throw exceptions in the parse-error case then my suggestion isnt necessary?